net/Socket.hpp

```diff
@@ -1360,8 +1360,8 @@ class StreamSocket : public Socket,
     /// Will always shutdown the socket.
     bool sendAndShutdown(http::Response& response);
 
-    /// Safely flush any outgoing data.
-    inline void flush()
+    /// Safely attempt to write any outgoing data.
+    inline void attemptWrites()
```

Comment:
Aha, flush() suggested all the outgoing buffer is written by the time we return here, while in fact this is just an attempt and there is no guarantee that that the outgoing buffer will be indeed empty by the time we return -- if I understand correctly.

Makes sense, then